oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2TokenEndpointFilterTests.java

```diff
@@ -363,6 +414,21 @@ private static MockHttpServletRequest createAuthorizationCodeTokenRequest(Regist
 		return request;
 	}
 
+	private static MockHttpServletRequest createAuthorizationCodeAndClientIdTokenRequest(RegisteredClient registeredClient) {
+		String[] redirectUris = registeredClient.getRedirectUris().toArray(new String[0]);
+
+		String requestUri = OAuth2TokenEndpointFilter.DEFAULT_TOKEN_ENDPOINT_URI;
+		MockHttpServletRequest request = new MockHttpServletRequest("POST", requestUri);
+		request.setServletPath(requestUri);
+
+		request.addParameter(OAuth2ParameterNames.GRANT_TYPE, AuthorizationGrantType.AUTHORIZATION_CODE.getValue());
```

Comment:
@vincent-hsin This request is invalid.

As per spec, in [Section 4.1.3.  Access Token Request](https://tools.ietf.org/html/rfc6749#section-4.1.3):

> If the client type is confidential or the client was issued client
   credentials (or assigned other authentication requirements), the
   client MUST authenticate with the authorization server as described
   in Section 3.2.1.

Only confidential clients are supported at the moment, so the client MUST authenticate at the token endpoint.

The change in this PR would allow a malicious user to obtain a token.

FYI, public client authentication support is coming in #45.

I'm going to close this PR and associated issue based on the above explanation.

Reply:
Ok, I got it. I will contribute for #45.